foundations/foundation-common/src/main/java/org/apache/servicecomb/foundation/common/utils/JvmUtils.java

```diff
@@ -39,24 +40,47 @@ private JvmUtils() {
 
   /**
    *
-   * @return main class or null, never throw exception
+   * @return main class or null, never throw exception.
+   * Note that this method does not ensure that the scbMainClass can be returned correctly in the some scene.
+   */
+  public static Class<?> findMainClassByStackTrace() {
+    String mainClass = null;
+    StackTraceElement[] stackTrace = new RuntimeException().getStackTrace();
+    if (stackTrace != null && stackTrace.length > 0) {
+      for (StackTraceElement stackTraceElement : stackTrace) {
+        if ("main".equals(stackTraceElement.getMethodName())) {
+          mainClass = stackTraceElement.getClassName();
+          break;
+        }
+      }
+    }
+    try {
+      Class<?> cls = Class.forName(mainClass);
```

Comment:
It's better to check if mainClass is null

Reply:
In the try code block, it will catch a null pointer exception and print the exception stack and the method returns null. I started to write the same way as you. Later, I saw my colleague's original writing and found that it can work normally. However, an unnecessary exception stack log may be generated.
```Java
try {
      Class<?> cls = Class.forName(mainClass);
      LOGGER.info("Found main class \"{}\" by stackTrace.", mainClass);
      return cls;
    } catch (Throwable e) {
      LOGGER.warn("\"{}\" is not a valid class.", mainClass, e);
      return null;
    }
```

Reply:
Yes. Do not use RuntimeException(e.g. NullPointerException) to do logic control. It's better to check null reference.